core/src/main/java/org/apache/brooklyn/core/mgmt/internal/LocalEntityManager.java

```diff
@@ -268,9 +270,26 @@ public void manage(Entity e) {
                     new Exception("source of duplicate management of "+e));
             return;
         }
+        // Validate all config before attempting to manage any entity. Do this here rather
+        // than in manageRecursive so that rebind is unaffected.
+        validateDescendantConfig(e);
```

Comment:
Only that this came first and the idea was to do validate constraints pre-management. You're right that it's better in the entity factory. As an added bonus, this makes an execution context available for the `getNonBlocking` suggestion.